pkg/pillar/containerd/oci.go

```diff
@@ -150,16 +143,52 @@ func (s *ociSpec) AddLoader(volume string) error {
 			if len(keyAndValueSlice) == 2 {
 				//handles Key=Value case
 				envContent = envContent + fmt.Sprintf("export %s=\"%s\"\n", keyAndValueSlice[0], keyAndValueSlice[1])
+				envWhitelist = append(envWhitelist, keyAndValueSlice[0])
 			} else {
 				//handles Key= case
 				envContent = envContent + fmt.Sprintf("export %s\n", e)
+				envWhitelist = append(envWhitelist, e)
 			}
-
 		}
 		if err := ioutil.WriteFile(filepath.Join(volumeRoot, "environment"), []byte(envContent), 0644); err != nil {
 			return err
 		}
 
+		envInsert := ""
+		if len(envWhitelist) > 0 {
+			envInsert = fmt.Sprintf("-w '%s'", strings.Join(envWhitelist, ","))
+		}
+
+		// create cmdline manifest
+		// each item needs to be independently quoted for initrd
+		execpathQuoted := make([]string, 0)
+		for _, s := range s.Process.Args {
+			execpathQuoted = append(execpathQuoted, fmt.Sprintf("\"%s\"", s))
+		}
+		execpath := strings.Join(execpathQuoted, " ")
+		username := ""
+		if s.Process.User.UID != 0 {
```

Comment:
Not a bit fan of this if statement -- can't you just use sudo that would allow specifying numeric ID directly. E.g.:

```
sudo -u \#65534 id
```

Reply:
Now it provides `{"severity":"info","source":"9cda1487-d017-4d8d-8770-cda0988f077b.1.2","content":"Executing /usr/bin/sudo -u '#1000' -g '#1000' --preserve-env=PATH,DISTTAG,FGC,FBR,HOME,password \"expect\" \"./startup.sh\"\n","msgid":3858,"timestamp":{"seconds":1613728080}}`